plugins/CoreVisualizations/Visualizations/Sparklines.php

```diff
@@ -202,9 +230,21 @@ private function fetchConfiguredSparklines()
 
                         list($compareValues, $compareDescriptions, $evolutions) = $this->getValuesAndDescriptions($compareRow, $columnToUse, '_change', '_trend');
 
+                        $metricsObjs = $report->getProcessedMetricsById();
                         foreach ($compareValues as $i => $value) {
+
+                            // Format metric
+                            $formatter = new MetricFormatter();
+                            $valueFormatted = $value;
+                            foreach ($metricsObjs as $metricObj) {
+                                if ((is_array($column) && in_array($metricObj->getName(), ($column)) || $metricObj->getName() == $column)) {
+                                    $valueFormatted = $metricObj->format($value, $formatter);
+                                    break;
+                                }
+                            }
```

Comment:
Finding the correct metric for formatting can be done outside the nested foreach loops, no need to do that over and over again. Also this is actually not always accurate. Their can be multiple columns given, which might not necessarily have the same format. e.g. it could be a percent and money value. Currently both metrics would be formatted with the first matching one. Guess we need to format each separately...